source/extensions/clusters/common/logical_host.cc

```diff
@@ -6,14 +6,14 @@ namespace Upstream {
 Upstream::Host::CreateConnectionData LogicalHost::createConnection(
     Event::Dispatcher& dispatcher, const Network::ConnectionSocket::OptionsSharedPtr& options,
     Network::TransportSocketOptionsConstSharedPtr transport_socket_options) const {
-  const auto current_address = address();
-  const std::vector<Network::Address::InstanceConstSharedPtr>& address_list = addressList();
-
+  const std::pair<Network::Address::InstanceConstSharedPtr,
```

Comment:
consider using 

`const auto [current_address, list] = addressAndListCopy();`

rather than creating a pair variable; then I think the call sites below don't change, and also I'd then my retract my request to make 'using' nicknames for the vector etc.

Reply:
Thanks for the comments! I will address these after we deemed the current approach is in the right direction.

Reply:
done